src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Target.java

```diff
@@ -203,4 +203,11 @@ public boolean hasSealedClasses() {
     public boolean obsoleteAccStrict() {
         return compareTo(JDK1_17) >= 0;
     }
+
+    /** Omit unused enclosing instance fields from inner classes that don't access enclosing
+     * instance state.
+     */
+    public boolean omitUnusedPrivateSyntheticEnclosingInstanceFields() {
```

Comment:
nit: what about a shorter name for this one?

Reply:
SGTM, maybe `optimizeOuterThis`? I'm open to suggestions :)